src/once_cell.rs

```diff
@@ -476,6 +476,7 @@ impl<T> OnceCell<T> {
     /// # });
     /// ```
     pub async fn get_or_init<Fut: Future<Output = T>>(&self, closure: impl FnOnce() -> Fut) -> &T {
+        #[allow(unreachable_patterns)]
```

Comment:
```suggestion
        #[allow(unreachable_patterns)] // false positive: https://github.com/rust-lang/rust/issues/129352
```
It would be nice to have a comment that shows why this is being ignored.

Reply:
Added